src/rebar_packages.erl

```diff
@@ -22,57 +25,93 @@
 -type vsn() :: binary().
 -type package() :: pkg_name() | {pkg_name(), vsn()}.
 
--spec packages(rebar_state:t()) -> ets:tid().
-packages(State) ->
-    catch ets:delete(?PACKAGE_TABLE),
-    case load_and_verify_version(State) of
-        true ->
-            ok;
-        false ->
+-spec packages(binary(), rebar_state:t()) -> ets:tid().
+packages(Registry, State) ->
+    case load_and_verify_version(Registry, State) of
+        {ok, T1} ->
+            {ok, T1};
+        _ ->
             ?DEBUG("Error loading package index.", []),
-            handle_bad_index(State)
+            handle_bad_index(Registry, State)
     end.
 
-handle_bad_index(State) ->
-    ?ERROR("Bad packages index. Trying to fix by updating the registry.", []),
-    {ok, State1} = rebar_prv_update:do(State),
-    case load_and_verify_version(State1) of
-        true ->
-            ok;
-        false ->
-            %% Still unable to load after an update, create an empty registry
-            ets:new(?PACKAGE_TABLE, [named_table, public])
+handle_bad_index(Registry, State) ->
+    ?ERROR("Bad packages index. Trying to fix by updating the registry from ~s", [Registry]),
+    case rebar_prv_update:do(Registry, State) of
+        {ok, State1} ->
+            case load_and_verify_version(Registry, State1) of
+                {ok, T} ->
+                    {ok, T};
+                false ->
+                    %% Still unable to load after an update, create an empty registry
+                    ets:new(repo, [public])
+            end;
+        E ->
+            throw(E)
     end.
 
-close_packages() ->
-    catch ets:delete(?PACKAGE_TABLE).
+close_packages(T) ->
+    catch ets:delete(T).
 
-load_and_verify_version(State) ->
-    {ok, RegistryDir} = registry_dir(State),
-    case ets:file2tab(filename:join(RegistryDir, ?INDEX_FILE)) of
-        {ok, _} ->
-            case ets:lookup_element(?PACKAGE_TABLE, package_index_version, 2) of
```

Comment:
No, it was used only during creation of the rebar3 formated registry. With this PR we use the hex registry unchanged.